tests/e2e-cypress/integration/menuTray.cy.js

```diff
@@ -0,0 +1,267 @@
+        category: 'layer',
+        id: id,
+        label: `Fake WMTS layer ${id}`,
+        layerBodId: 'test.wmts.layer.' + id,
+        staging: 'prod',
+    }
+}
+
+function addFakeWMTSLayer(layers, id) {
+    const layerName = 'test.wmts.layer.' + id
+    layers[layerName] = {
+        opacity: 1.0,
+        attribution: 'attribution.test.wmts.layer',
+        background: false,
+        searchable: true,
+        format: 'png',
+        queryableAttributes: ['id', 'name'],
+        topics: 'ech,test-topic-standard,test-topic-with-active-and-visible-layers',
+        attributionUrl: 'https://api3.geo.admin.ch/',
+        tooltip: true,
+        timeEnabled: false,
+        highlightable: true,
+        chargeable: false,
+        timestamps: ['current'],
+        hasLegend: true,
+        label: 'WMTS test layer ' + id,
+        type: 'wmts',
+        serverLayerName: 'test.wmts.layer.' + id,
+    }
+    return layers
+}
+
+function getFixturesAndIntercepts(nbLayers, nbSelectedLayers) {
+    const topicId = 'ech'
+    let fixAndInt = getDefaultFixturesAndIntercepts()
+    fixAndInt.addCatalogFixtureAndIntercept = () => {
+        let layersCatalogEntries = []
+        for (let i = 2; i < nbLayers + 2; i++) {
+            layersCatalogEntries = layersCatalogEntries.concat(getFakeWMTSLayerCatalogEntry(i))
+        }
+        const catalog = {
+            results: {
+                root: {
+                    category: 'root',
+                    staging: 'prod',
+                    id: 1,
+                    children: layersCatalogEntries,
+                },
+            },
+        }
+        // intercepting further topic metadata retrieval. Each topic has its own topic tree
+        cy.intercept(`**/rest/services/${topicId}/CatalogServer?lang=**`, {
+            body: catalog,
+        }).as(`topic-${topicId}`)
+    }
+    fixAndInt.addLayerFixtureAndIntercept = () => {
+        let layers = {}
+        for (let i = 2; i < nbLayers + 2; i++) {
+            addFakeWMTSLayer(layers, i)
+        }
+        cy.intercept('**/rest/services/all/MapServer/layersConfig**', {
+            body: layers,
+        }).as('layers')
+    }
+    fixAndInt.addTopicFixtureAndIntercept = () => {
+        let activatedLayers = []
+        for (let i = 2; i < nbSelectedLayers + 2; i++) {
+            activatedLayers.push('test.wmts.layer.' + i)
+        }
+        cy.intercept('**/rest/services', {
+            body: {
+                topics: [
+                    {
+                        activatedLayers,
+                        backgroundLayers: ['test.background.layer', 'test.background.layer2'],
+                        defaultBackground: 'test.background.layer2',
+                        groupId: 1,
+                        id: topicId,
+                        plConfig: null,
+                        selectedLayers: [],
+                    },
+                ],
+            },
+        }).as('topics')
+    }
+    return fixAndInt
+}
```

Comment:
The problem is that by the time "goToMapView" finishes, it would already be to late to add new intercepts, as the correct intercepts are already needed during initialization.